lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -71,81 +71,131 @@ void SetMessageManagerHandler::SetAmfSessionContext(
        * intial message. Check if it's initial message
        */
-      if ((cfg.rat_specific_context.m5gsm_session_context().rquest_type() ==
+      if ((cfg.rat_specific_context.m5gsm_session_context().request_type() ==
            INITIAL_REQUEST) &&
           (cfg.common_context.sm_session_state() == CREATING_0)) {
         /* it's new UE establisment request and need to create the session
          * context
          */
-        MLOG(MINFO)
-            << "AMF request type INITIAL_REQUEST and session state CREATING";
         auto session_map = session_store_.read_sessions({imsi});
-        send_create_session(session_map, imsi, session_id, cfg, dnn);
+        send_create_session(session_map, imsi, cfg, pdu_id);
+        response_callback(Status::OK, SmContextVoid());
+        return;
       }
+      MLOG(MERROR)
+          << "AMF request type- Unhandled request type:"
+          << cfg.rat_specific_context.m5gsm_session_context().request_type();
+
+      Status status(grpc::UNKNOWN, "Unknown session state or request");
+      response_callback(status, SmContextVoid());
+    }
+    return;
+  });
+}
+
+/* Handling set message from AMF
+ * check if PDU_SESSION exists
+ * if the PDU_SESSION doen't exists, log and ignore
+ * If EXISTING_PDU_SESSION, get the session entry, check on incoming session
+ * state and version  accordingly take action, write to memory by SessionStore
+ */
+
+void SetMessageManagerHandler::SetSmfNotification(
+    ServerContext* context, const SetSmNotificationContext* notif,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  // Print the message from AMF
+  PrintGrpcMessage(static_cast<const google::protobuf::Message&>(*notif));
+  /* Read the event type from the prot message
+   */
+  auto& noti = *notif;
+  m5g_enforcer_->get_event_base().runInEventBaseThread([this, response_callback,
+                                                        noti]() {
+    NotifyUeEvents Uevent = noti.rat_specific_notification().notify_ue_event();
+    MLOG(MINFO) << "Notification of imsi: " << noti.common_context().sid().id()
+                << " from AMF  Event value:" << Uevent;
+    switch (Uevent) {
+      case PDU_SESSION_INACTIVE_NOTIFY:
+        pdu_session_inactive(noti, response_callback);
+        return;
+      case UE_IDLE_MODE_NOTIFY:
+        idle_mode_change_sessions_handle(noti, response_callback);
+        return;
+      default:
+        return;
     }
-    response_callback(Status::OK, SmContextVoid());
   });
 }
 
 /* Creeate respective SessionState and context*/
 void SetMessageManagerHandler::send_create_session(
-    SessionMap& session_map, const std::string& imsi,
-    const std::string& session_id, const SessionConfig& cfg,
-    const std::string& dnn) {
-  /* If it is new session to be created, check for same DNN exists
-   * for same IMSI, i.e if IMSI found and respective DNN found in
+    SessionMap& session_map, const std::string& imsi, SessionConfig& cfg_new,

```

Comment:
`cfg_new` -> `new_cfg` (reads more like english!)

Reply:
Replace the parameter name from cfg_new -> new_cfg as suggested.